src/runtime-rs/crates/runtimes/virt_container/src/lib.rs

```diff
@@ -20,15 +20,19 @@ use agent::{kata::KataAgent, AGENT_KATA};
 use anyhow::{anyhow, Context, Result};
 use async_trait::async_trait;
 use common::{message::Message, RuntimeHandler, RuntimeInstance};
-use hypervisor::{dragonball::Dragonball, Hypervisor, HYPERVISOR_DRAGONBALL};
+#[cfg(not(target_arch = "s390x"))]
+use hypervisor::{dragonball::Dragonball, HYPERVISOR_DRAGONBALL};
+use hypervisor::Hypervisor;
 use hypervisor::{qemu::Qemu, HYPERVISOR_QEMU};
+#[cfg(not(target_arch = "s390x"))]
+use kata_types::config::DragonballConfig;
 use kata_types::config::{
-    hypervisor::register_hypervisor_plugin, DragonballConfig, QemuConfig, TomlConfig,
+    hypervisor::register_hypervisor_plugin, QemuConfig, TomlConfig,
 };
 
-#[cfg(feature = "cloud-hypervisor")]
+#[cfg(all(feature = "cloud-hypervisor", not(target_arch = "s390x")))]
 use hypervisor::ch::CloudHypervisor;
-#[cfg(feature = "cloud-hypervisor")]
+#[cfg(all(feature = "cloud-hypervisor", not(target_arch = "s390x")))]
```

Comment:
I would love for the resulting code to have fewer conditional compilation lines but frankly, I don't see how to achieve that at the moment, without a major refactor to group individual hypervisors' code more closely together.